src/main/java/org/springframework/shell2/JLineShell.java

```diff
@@ -155,10 +165,7 @@ public void run() throws IOException {
 					result = ReflectionUtils.invokeMethod(method, methodTarget.getBean(), args);
 				}
 				catch (ExitRequest e) {
-					if (applicationContext instanceof Closeable) {
```

Comment:
Should we make the ExitRequest an ExitResult instead? And register an ExitResultHandler by default that will close the app? (I appreciate it's not directly related to this PR :) )